File: experiments/python_rewrite/seed_map_layer/seed_map_layer.py

from seed_map.seed_map import SeedMap

class SeedMapLayer:

    def __init__(self):
        # seeds to be a seed_map
        self.seed_maps:SeedMap = []
        pass

    def add_map(self, seed_map):
        self.seed_maps.append(seed_map)
        pass

    def sort_maps(self):
        self.seed_maps = sorted(self.seed_maps, key=lambda x: x.source)

    def map_seed(self, value: int):

        for seed_map in self.seed_maps:
            if (result:= seed_map.map_seed(value)) is not None:
                return result
        
        return value
    
    def map_seed_inverse(self, value):
        for seed_map in self.seed_maps:
            if(result:= seed_map.map_seed_inverse(value)) is not None:
                return result
            
        return value

File: 2023/day05/java/day5/src/main/java/com/mycompany/app/SeedMapLayer.java
package com.mycompany.app;

import java.util.ArrayList;
import java.util.Optional;

public class SeedMapLayer {
    
   private ArrayList<SeedMap> seed_maps;

    public SeedMapLayer()
    {
        seed_maps = new ArrayList<SeedMap>();
    }

    public void AddSeedMap(SeedMap seed_map)
    {
        seed_maps.add(seed_map);
    }

    public long MapSeed(long seed_value)
    {
        for (SeedMap seed_map : seed_maps) {
            Optional<Long> result = seed_map.MapSeed(seed_value);

            if (result.isPresent()) return result.get(); 
        }

        return seed_value;
    }
}
